support-files/mariadb-execstartpre-check.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+#
+# This script purpose is to check if MariaDB
+# configuration has problematic configurations
+# and make more reasonable error messages than
+# it does now
+#
+# To make this script work you need:
+#   * my_print_defaults from MariaDB package
+#   * systemctl from systemd package
+#
+MY_PRINT_DEFAULTS=$(which my_print_defaults)
+
+if [ -n "${MY_PRINT_DEFAULTS}" ]
+then
+    echo "Can't find tool: my_print_defaults"
+fi
+
+#
+# Check with my_print_defaults that any
+# path in configuration is not pointing somewhere
+# we can't write or it is invisible
+#
+# Check with systemctl cat mariadb*.service
+# if user has already altered service file to
+# allow writing to there locations
+#
+my_print_defaults --mysqld | while read -r setting
+do
+    conf_var=${setting%%=*}
+    conf_var=${conf_var#--}
+    conf_val=${setting#*=}
+    conf_problem=""
+
+    # These are the places that systemd
+    # is prevented currently to write or access
+    case "$conf_val" in
+        # ProtectSystem prevents writes to /usr, /boot and /efi
+        /usr*|/boot*|/efi*)
+            conf_problem="ProtectSystem"
+            ;;
+        # ProtectHome prevents access to /home, /root and /run/user
+        /home*|/root*|/run/user*)
```

Comment:
What you can do to determine if there is a problem is the bash `[ -w "${conf_val}" ]` to determine if writeable.

I tested with:
```
$ sudo systemctl status bob.service
× bob.service
     Loaded: loaded (/etc/systemd/system/bob.service; static)
     Active: failed (Result: exit-code) since Thu 2021-09-23 12:26:36 AEST; 8s ago
    Process: 161902 ExecStartPre=sh -c [ -w /home/dan ] && echo I can write (code=exited, status=1/FAILURE)
        CPU: 10ms

Sep 23 12:26:36 localhost.localdomain systemd[1]: Starting bob.service...
Sep 23 12:26:36 localhost.localdomain systemd[1]: bob.service: Control process exited, code=exited, status=1/FAILURE
Sep 23 12:26:36 localhost.localdomain systemd[1]: bob.service: Failed with result 'exit-code'.
Sep 23 12:26:36 localhost.localdomain systemd[1]: Failed to start bob.service.

$ cat /etc/systemd/system/bob.service
[Service]
ProtectHome=true

ExecStartPre=sh -c '[ -w /home/dan ] && echo I can write'
ExecStart=sleep 2
```